hibernate-core/src/main/java/org/hibernate/relational/SchemaManager.java

```diff
@@ -94,4 +94,28 @@ public interface SchemaManager extends jakarta.persistence.SchemaManager {
 	 */
 	@Incubating
 	void populate();
+
+	/**
+	 * Obtain an instance which targets the given schema.
+	 * <p>
+	 * This is especially useful when using multiple schemas, for example, in
+	 * {@linkplain org.hibernate.cfg.MultiTenancySettings#MULTI_TENANT_SCHEMA_MAPPER
+	 * schema-based multitenancy}.
+	 *
+	 * @param schemaName The name of the schema to target
+	 *
+	 * @since 7.1
+	 */
+	@Incubating
+	SchemaManager forSchema(String schemaName);
```

Comment:
Would `forTenant` make sense, maybe?

Reply:
It would have to know how to use the `MultiTenantConnectionProvider` though.